pkg/logger/logger.go

```diff
@@ -0,0 +1,139 @@
+package logger
+
+import (
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+/*
+This package wraps the zap logging API for use across services without passing a shared logger.
+Fields can be added to a requestID using the WriteFields method, these will be added to any
+subsequent XxxWithID log calls. To preserve performance ClearFields must be called when the
+requestID's thread is closed as a sync map is used internally.
+Child loggers can be created from a parent logger using the WithFields method, this child logger
+will append the provided fields to all logs, whilst maintaining a reference to the top level
+request fields pool.
+
+Example:
+
+WrappedLogger := NewLogger(myLogger)
+WrappedLogger.WriteFields("my-id", zap.String("foo", "bar"))
+WrappedLogger2 := WrappedLogger.WithFields(zap.String("ping", "pong"))
+
+WrappedLogger.DebugWithID("my-id", "my log line")
+	=> {"level":"debug","foo":"bar","msg":"my log line""}
```

Comment:
Should `"id":"my-id"` be present in these examples?

Reply:
resolved